test/test_dofs.jl

```diff
@@ -504,3 +504,154 @@ end
+        ],
+
+        # Component coupling
+        [
+            true    true    true    true
+            true    true    true    true 
+            true    true    true    true
+            true    true    true    true 
+        ],
+        [
+            true     false    false    false
+            false    true     false    false 
+            false    false    true     false
+            false    false    false    true 
+        ],
+        [
+            true    true    true    false
+            true    true    true    true 
+            true    true    true    true
+            false    true    true    true 
+        ],
+    ]
+    function is_stored(A, i, j)
+        A = A isa Symmetric ? A.data : A
+        for m in nzrange(A, j)
+            A.rowval[m] == i && return true
+        end
+        return false
+    end
+
+    function check_coupling(dh, coupling_dofs, elements_coupling_dofs, topology, K, coupling, elements_coupling)
+        for cell_idx in eachindex(getcells(dh.grid))
+            fh = dh.fieldhandlers[dh.cell_to_fieldhandler[cell_idx]]
+            # Test in-element coupling
+            for (index, (i_dofs, j_dofs, i_name, j_name)) in pairs(coupling_dofs)
+                field1_idx = findfirst(i->i.name == i_name,fh.fields)
+                field2_idx = findfirst(i->i.name == j_name,fh.fields)
+                any(isnothing.([field1_idx, field2_idx])) && continue
+                for i_idx in i_dofs, j_idx in j_dofs
+                    i = dh.cell_dofs[i_idx + dh.cell_dofs_offset[cell_idx] - 1]
+                    j = dh.cell_dofs[j_idx + dh.cell_dofs_offset[cell_idx] - 1]
+                    @test is_stored(K, i, j) == coupling[index]
+                end
+            end
+            # test cross-element coupling
+            for neighbor_idx in topology.cell_face_neighbor[cell_idx]
+                fh_neighbor = dh.fieldhandlers[dh.cell_to_fieldhandler[neighbor_idx.idx]]
+                for (index, (i_dofs, j_dofs, i_name, j_name)) in pairs(elements_coupling_dofs)
+                    field_idx = findfirst(i->i.name == i_name,fh.fields)
+                    neighbor_field_idx = findfirst(i->i.name == j_name,fh_neighbor.fields)
+                    any(isnothing.([field_idx, neighbor_field_idx])) && continue
+                    ip = fh.fields[field_idx].interpolation
+                    ip_neighbor = fh_neighbor.fields[neighbor_field_idx].interpolation
+                    is_discontinuous =  all(Ferrite.IsDiscontinuous.([ip, ip_neighbor]))
+                    for i_idx in i_dofs, j_idx in j_dofs
+                        i = dh.cell_dofs[i_idx + dh.cell_dofs_offset[cell_idx] - 1]
+                        j = dh.cell_dofs[j_idx + dh.cell_dofs_offset[neighbor_idx.idx] - 1]
+                        !is_discontinuous && continue
+                        @test is_stored(K, i, j) == (elements_coupling[index])
+                    end
+                end
+            end
+        end
+    end
```

Comment:
I think this now acts like the other tests but somewhat automated * and hopefully doesn't mirror internals *

Reply:
Why not use `celldofs` instead of `dh.cell_dofs`?

Reply:
Missed it, will change it with next push. 😄